experimental/packages/opentelemetry-instrumentation/src/types.ts

```diff
@@ -43,6 +44,9 @@ export interface Instrumentation {
   /** Method to set meter provider  */
   setMeterProvider(meterProvider: MeterProvider): void;
 
+  /** Method to set logger provider  */
+  setLoggerProvider(loggerProvider: LoggerProvider): void;
```

Comment:
nit: To support backward compatibility, shoudn't this be marked as optional. So anyone currently implementing this interface won't be broken and it will still compile.

Downside is that all usages would need to check if this method exists.

Reply:
And yes, the above comment "assumes" that people are not necessarily using the InstrumentationBase.

Reply:
Added method as optional